src/vm/win32threadpool.cpp

```diff
@@ -1815,8 +1815,8 @@ Thread* ThreadpoolMgr::CreateUnimpersonatedThread(LPTHREAD_START_ROUTINE lpStart
         // CreateNewThread takes care of reverting any impersonation - so dont do anything here.
         bOK = pThread->CreateNewThread(0,               // default stack size
                                        lpStartAddress,
-                                       lpArgs           //arguments
-                                       );
+                                       lpArgs,           //arguments
+                                       W("ThreadPool"));
```

Comment:
> The only thing I have not found are where server GC threads are created

This is in `src\gc\gc.cpp`, in the function `gc_heap::create_gc_thread`, which ultimately delegates to here: https://github.com/dotnet/coreclr/blob/master/src/vm/gcenv.os.cpp#L636. You could add an optional name parameter to `GCToOSInterface::CreateThread` and do the actual thread naming there.

If you go down this route for this PR, you'll also need to change the signature of the implementations https://github.com/dotnet/coreclr/blob/master/src/gc/windows/gcenv.windows.cpp#L555 (Windows-only) and https://github.com/dotnet/coreclr/blob/master/src/gc/unix/gcenv.unix.cpp#L702 (POSIX-y platforms only). I can give you some more guidance on that if you want to do this as a part of this PR - otherwise you can create an issue and I'll follow-up on it later. (These files don't build as part of a normal build - it's only when building the GC as a standalone library.)